sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -555,6 +557,100 @@ case class ArraySort(child: Expression) extends UnaryExpression with ArraySortLi
   override def prettyName: String = "array_sort"
 }
 
+
+/**
+ * Returns a random permutation of the given array..
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(array) - Returns a random permutation of the given array.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(array(1, 20, 3, 5));
+       [3, 1, 5, 20]
+      > SELECT _FUNC_(array(1, 20, null, 3));
+       [20, null, 3, 1]
+  """, since = "2.4.0")
+case class Shuffle(child: Expression) extends UnaryExpression with ImplicitCastInputTypes {
```

Comment:
Isn't `ExpectsInputTypes` sufficient in this case?

Reply:
Correct. Input is an Array. No string for input. Fixed.